homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -79,22 +254,66 @@ def __init__(
         self._processors: list[PassiveBluetoothDataProcessor] = []
         self._update_method = update_method
         self.last_update_success = True
+        self.restore_data: dict[str, dict[str, Any]] = {}
+        self.restore_key = None
+        if config_entry := config_entries.current_entry.get():
+            self.restore_key = config_entry.entry_id
+        self._on_stop.append(async_register_coordinator_for_restore(self.hass, self))
 
     @property
     def available(self) -> bool:
         """Return if the device is available."""
         return super().available and self.last_update_success
 
+    @callback
+    def async_get_restore_data(
+        self,
+    ) -> dict[str, dict[str, Any]]:
+        """Generate the restore data."""
+        return {
+            processor.restore_key: processor.data.async_get_restore_data()
+            for processor in self._processors
+            if processor.restore_key
+        }
+
+    @callback
+    def _async_start(self) -> None:
+        """Start the callbacks."""
+        super()._async_start()
+        hass = self.hass
+        # If Home Assistant is already running we need to restore the
+        # last service info as well since the startup restore has already
+        # happened.
+        if hass.is_running and (
+            last_service_info := async_last_service_info(
+                hass, self.address, self.connectable
+            )
+        ):
+            self._async_handle_bluetooth_event(
+                last_service_info, BluetoothChange.ADVERTISEMENT
+            )
```

Comment:
This is a different problem and should get its own PR.  Currently we load the data at startup but if the integration reloads it will not get the last service info

Reply:
This was an order problem and not an actual bug